packages/flutter_tools/test/general.shard/android/gradle_errors_test.dart

```diff
@@ -0,0 +1,698 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+
+import 'package:flutter_tools/src/android/gradle_utils.dart';
+import 'package:flutter_tools/src/android/gradle_errors.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/reporting/reporting.dart';
+import 'package:mockito/mockito.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/mocks.dart';
+
+void main() {
+  group('gradleErrors', () {
+    test('list of errors', () {
+      // If you added a new Gradle error, please update this test.
+      expect(gradleErrors,
```

Comment:
What sort of breakage does this test prevent?

Reply:
Accidentally removing an error handler or changing the order of the handlers since that can affect what error is recognized.

Reply:
Ah, okay. If the matchers can match more than one error, it would be good to document somewhere why the current ordering was chosen, or otherwise any help you can give to future tool engineers who are trying to decide how to modify the list. How will they answer the question, "Where do I put a new element in the list?" You can also include your response to my initial question as a comment on the test.

Reply:
Yeah. I added a comment